minifi-c2/minifi-c2-commons/src/main/java/org/apache/nifi/minifi/c2/model/FlowStatus.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.c2.model;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.Map;
+
+/**
+ * Status of the aspects of the flow that are controllable by the C2 server, ie:
+ *   - Processors that can be started/stopped and their current state
+ *   - Queues that can be cleared and their current state
+ */
+@ApiModel
+public class FlowStatus {
```

Comment:
Do we intent to have an analog for non flow component statuses?

Reply:
Not sure I follow. Do you have an example of something that might not into the proposed model?

Reply:
Perhaps it is the naming, but the name  FlowComponentStatus implies that it is part of the flow, but there may be components that run that are flow independent. For example think of things that may run a procedure. That component could be tracked in the FlowComponentStatus map components, but perhaps not. 

Additionally, not sure I mentioned this already but I see that tracking maps are not concurrent map impls. Should I assume that state is monitored/controlled elsewhere?

Reply:
OK, I got it. The current structure assumes to much and is limiting. Will take another look at that, thanks.

Yes, I don't think these DTOs need to worry about concurrent access. Official model and state and would be managed at a lower level in the C2 server (persisted or cached and accessed through thread safe interfaces). These objects represent a snapshot copy for I/O. It should become clearer in my next PR that adds some of the service and persistence layer foundations.